### PR TITLE
Docs: JSONata refresh and update with new examples

### PR DESCRIPTION
**Pull request by oatybiscuit**
Following your message last November, here is my submission:
[Do you have any interest in writing a JSONata guide geared toward using it with Home Assistant?](https://community.home-assistant.io/t/node-red-passing-variables-into-call-service-node/617410/7?u=biscuit)

Refresh JSONata material in the documentation.
Updated the Guide page with table to cross reference JSONata use to Nodes to Examples
Added JSONata 'primer' guide page
Reorganised JSONata examples to focus on node and use-type
Refresh existing examples and added new ones, with additional explanation on how to use JSONata specifically at each point.

All documentation is my own original work and I have not used any AI, only cross referencing back to the WebSocket documentation and the JSONata documentation.

All examples have been fully tested (in my environment) and the code checked.
All links updated and checked as far as I am able to.
New pages have been added to the vue-doc configuration (but have not been fully tested).

Unfortunately I live in the UK, and am currently on GMT which is the same as UTC so I am unable to double check any issues that may arise from the use of sensor.time_utc or sensor.time in regard to other sensor timestamps (local - UTC issues).

### CODE REVIEW

**Comment by zachowj** [comment]
Wow, very nice.

File: docs/guide/jsonata-primer.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+The `$spread()` function takes a JSON object and returns an array of objects, each with a single key:value pair.
+The `$merge()` function reverses this, taking an array of objects and combining back into one object. A particular feature of this function is that, where key values are duplicated, the end result contains only the _last_ such key to be found.
+Therefore, the approach to take is to spread the object, append the replacement key:value as an object to the end of this array, then to merge the array back together.
+
+    (
+      $object:={"first": 1, "second": 2, "third": 3};
+      $replaceKey:= "second";
+      $replaceVal:= 15;
+      $spread($object) ~> $append({$replaceKey: $replaceVal}) ~> $merge()
+    )
```
Comment (zachowj) [inline docs/guide/jsonata-primer.md:199]:
```suggestion
   ```jsonata
    (
      $object:={"first": 1, "second": 2, "third": 3};
      $replaceKey:= "second";
      $replaceVal:= 15;
      $spread($object) ~> $append({$replaceKey: $replaceVal}) ~> $merge()
    )
    ```
```

File: docs/guide/jsonata-primer.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+- and then using `$merge()` to merge this array of objects back into one object
+
+The simplicity comes from the power of JSONata to map, filter, sort and aggregate, however compared to most procedural languages it is a new way of thinking. Since the JSON object and variables cannot be modified, there is the question of how to, for example, change just one element in an array. The expression `$array[2]:=10` will return an error since the left hand side of assignment can _only_ be a variable and not an expression. To achieve this requires a _declaration_ of the new array, built as follows.
+
+    (
+      $array:=[1, 2, 3, 4, 5, 6];
+      $newval:=10;
+      $index:=2;
+      $append($array[[0..$index-1]], $newval)~>$append($array[[$index+1..$count($array)]])
+    )
```
Comment (zachowj) [inline docs/guide/jsonata-primer.md:180]:
```suggestion
    ```jsonata
    (
      $array:=[1, 2, 3, 4, 5, 6];
      $newval:=10;
      $index:=2;
      $append($array[[0..$index-1]], $newval)~>$append($array[[$index+1..$count($array)]])
    )
    ```
```

File: docs/cookbook/jsonata-1-call-service.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+- a domain (integration platform) such as 'climate'
+- a service call within this domain, such as 'set_temperature'
+- a target entity or entities (or area or device containing entities) to apply the service to
+- a JSON _data object_, containing any additional required and optional parameters
+
+The _data object_ will vary, depending on the integration and service call, from an empty object `{}` to something much more complex. Constructing this data object correctly is key to a successful service call, and **JSONata** rather than mustache templating should ideally be used to achieve this.
+
+**Note:** The data field used in a service call must be a valid JSON object.
+
+- Mustache templates are delimited with '{{msg.payload}}'. Although acceptable in most JSON input fields, this is _invalid syntax_ in JSONata expressions and must not be used.
```
Comment (zachowj) [inline docs/cookbook/jsonata-1-call-service.md:16]:
```suggestion
- Mustache templates are delimited with <code v-pre>{{msg.payload}}</code>. Although acceptable in most JSON input fields, this is _invalid syntax_ in JSONata expressions and must not be used.
```

**Review by zachowj — CHANGES_REQUESTED** [review]
Not sure if you were able to set up a local environment to test with but I just added [documentation](https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/documentation.html) on how to do this.

**Comment by oatybiscuit** [comment]
> Not sure if you were able to set up a local environment to test with but I just added [documentation](https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/documentation.html) on how to do this.

Thanks. Entirely new to Github, markdown, and Visual Studio so it may take me some time to get my head around setting up the local environment stuff. I assumed that I had not got everything correct but still hope it is mostly there!

**Pull request merged**